workers/unity/Assets/Gdk/Core/Error/IErrorDispatcher.cs

```diff
@@ -0,0 +1,9 @@
+namespace Improbable.Gdk.Core
+{
+    public interface IErrorDispatcher
+    {
+        void LogError(string message);
+
+        void LogErrorFormat(string format, params object[] args);
```

Comment:
I don't think there has been any discussion as to what kind of dispatcher we want, or how it should be exposed.

I also don't think providing string formatting is the kind of thing an error dispatcher really cares about. String formatting is a native things to C# anyway.
You need something which specifies an error well enough that the callback can decide what to do with it (log locally, log to spatial, quit, try to fix it in some rare case, set some flag somewhere else etc...)
A string message is useful for logging, but you likely want to give more information than that.

Reply:
Martin mentioned this to me yesterday. While I agree that an dispatcher shouldn't care about it, it mirros Unity's `Debug` class which makes it simple for the user.
In the end it depends on what we want to do with the dispatcher. Should it be a "logging dispatcher" or an "error handling dispatcher".  I would argue that we will actually need both to facilitate sending logging messages to spatial and to handle errors in a nice way,

Reply:
The reasoning for having LogErrorFormat was that we use the method a lot in our code with Debug and I felt that it would be ugly to wrap everything in string.Format and only have LogError.

I would say that we primarily want this to be a logging error dispatcher. If we can handle an error an error object should be returned instead.

Reply:
This is how we did it in the SDK. This does not work completely as it won't allow us to add additional parameters (e.g. entity id) to it. We basically lose all context.

I don't think there is a reason for having a logging error dispatcher. Either it's a generic logging dispatcher or it's actually handling the error in some way

Reply:
Ok I think it will be best if I add a few possible implementations into the doc and we can decide based on that.

Reply:
The point of this isn't to be like Unity's debug class (unless we decide that is needed). It's to generally register that there should be an error. A string doesn't even need to be the main part of that api, it just needs the ability to send some sort of object that describes an error. Having multiple overloaded for string formatting seems odd.

Reply:
ok, we want a dispatcher that will allow the user to handle an error in some way. This could be printing an error message, but this could also be some kind of retry-logic. in that case I agree that this shouldn't mirror Unity's debug class

Reply:
Though we will need to extend the current logging at some point to be able to send debugging statements to spatial (not as part of this pr) while still logging inside the editor

Reply:
You also don't need to use string.format (although most IDEs seem to prefer it to LogFormat). You can use `$"some formatted string with {someVariable} in it"`.

Reply:
We do want the default dispatcher to just log though.

Reply:
@samiwh @jessicafalk I have added two proposed solutions into the original design doc. One of them is the current one and the second uses Error Objects similar to the ones which are returned by worker.Connect()

Reply:
Ok I have implemented the error handler to use the IErrorContext interface. Some errors were very specific so I didnt not create the struct in a separate file because they were only used in one spot. For the others I created them separately.